### PR TITLE
[Misc] Add a script to install dependencies for the prover independent of the repo

### LINKED ISSUE #7148: [Feature Request] Make prover tool installation independent of aptos-core
Currently, as described in https://aptos.dev/cli-tools/install-move-prover, one needs to clone the aptos repo to install prover tools. That is not a good UX. We need to make those tools installable independent of the repo.

@rahxephon89 @junkil-park

### PR DESCRIPTION
**Pull request by rahxephon89**
### Description

As the first step to resolve #7148, this PR adds a script to install dependencies of the Move prover (`.net`, `z3`, `cvc5` and`boogie`) and `graphviz` without relying on the `aptos-core` repo. It can be run outside the repo. The usage of this script is similar to `dev_setup.sh` because the code is borrowed from it. 

### Test Plan

Need to test this script in different OS environments.

### CODE REVIEW

**Review by clay-aptos — APPROVED** [review]
Thank you, Teng! I will work with you to update the Move Prover install docs when this is merged.

**Review by wrwg — APPROVED** [review]
Nice!

File: developer-docs-site/static/scripts/prover_setup.sh [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+This script will download and install the necessary dependencies needed to
+build, test and inspect Aptos Core.
+
+Based on your selection, these tools will be included:
+EOF
+
+  if [[ "$INSTALL_PROVER" == "true" ]]; then
+cat <<EOF
+Move prover tools (since -y was provided):
```
Comment (junkil-park) [inline developer-docs-site/static/scripts/prover_setup.sh:218]: @rahxephon89 , perhaps, we don't need `-y` option? The purpose of running this script is to install the Prover dependency?

Comment (rahxephon89) [reply developer-docs-site/static/scripts/prover_setup.sh:218]: Thanks for the review Junkil! If -yp is not specified, the script will install all dependencies by default.

**Comment by clay-aptos** [comment]
Thanks, @rahxephon89 ! @saharct , can you help update the Move Prover install docs to account for this script?
https://aptos.dev/cli-tools/install-move-prover

Thank you both!

**Comment by rahxephon89** [comment]
@clay-aptos @saharct I will discuss with @wrwg on how to use this script and then submit a PR to update the install doc.

**Comment by clay-aptos** [comment]
Thank you so much, @rahxephon89 !

**Comment by saharct** [comment]
Thank you, @rahxephon89!

**Pull request merged**